### PR TITLE
feat: swap pairing for blst

### PR DESCRIPTION
**Pull request by cryptonemo**
Requires many other PRs merged.

Tracking issue: https://github.com/filecoin-project/rust-fil-proofs/issues/1438

### CODE REVIEW

**Comment by cryptonemo** [comment]
Updates to the latest Proofs `v9.0.1`

Resolves https://github.com/filecoin-project/rust-fil-proofs/issues/1438

**Comment by cryptonemo** [comment]
Resolves https://github.com/filecoin-project/rust-fil-proofs/issues/1484

File: README.md [hunk 1]
```diff
@@ -40,14 +42,16 @@ rm .install-filcrypto \
...
 ```
 
-By default, a 'gpu' option is used in the proofs library.  This feature is also used in FFI unless explicitly disabled.  To disable building with the 'gpu' dependency, set `FFI_USE_GPU=0`:
+By default, a 'gpu' option is used in the proofs library*.  This feature is also used in FFI unless explicitly disabled.  To disable building with the 'gpu' dependency, set `FFI_USE_GPU=0`:
 
 ```shell
 rm .install-filcrypto \
     ; make clean \
-    ; FFI_BUILD_FROM_SOURCE=1 FFI_USE_BLST=1 FFI_USE_GPU=0 make
+    ; FFI_BUILD_FROM_SOURCE=1 FFI_USE_PAIRING=1 FFI_USE_GPU=0 make
```
Comment (vmx) [inline README.md:50]: Why using `FFI_USE_PAIRING=1`? Is it to demonstrate that this option exists?

File: install-filcrypto [hunk 2]
```diff
@@ -215,7 +221,11 @@ get_release_flags() {
     fi
 
     # Maps cpu flag to rust flags (related to entries in rust/rustc-target-features-optimized.json)
-    feature_map=("adx:+adx" "sha_ni:+sha" "sha2:+sha2" "sse2:+sse2" "avx2:+avx2" "avx:+avx" "sse4_2:+sse4.2" "sse4_1:+sse4.1")
+    if [ "$(uname -s)" = "Darwin" ] && [ "$(uname -m)" = "x86_64" ]; then
+        feature_map=("adx:+adx" "sha_ni:+sha" "sha2:+sha2" "avx2:+avx2" "sse4_2:+sse4.2" "sse4_1:+sse4.1")
```
Comment (vmx) [inline install-filcrypto:225]: I guess there is a reason to not use `sse` and `avx` on MacOS x86?

File: rust/Cargo.toml [hunk 3]
```diff
@@ -32,14 +32,14 @@ rand = "0.7"
 rand_chacha = "0.2.1"
 rayon = "1.2.1"
 anyhow = "1.0.23"
-bellperson = { version = "0.14.1", default-features = false }
+bellperson = { version = "0.16", default-features = false }
 serde_json = "1.0.46"
 rust-gpu-tools = "0.3.0"
```
Comment (vmx) [inline rust/Cargo.toml:37]: `rust-gpu-tools` should also be updated to the version `bellperson` is using.

File: README.md (hunk 1 above)
Comment (cryptonemo) [reply README.md:50]: That's telling you how to enable pairing, now that it's not default.

File: install-filcrypto (hunk 2 above)
Comment (cryptonemo) [reply install-filcrypto:225]: They aren't supported for some reason (in the cross compile).

File: rust/Cargo.toml (hunk 3 above)
Comment (cryptonemo) [reply rust/Cargo.toml:37]: Good catch, thanks

File: README.md (hunk 1 above)
Comment (cryptonemo) [reply README.md:50]: I see, it was in the wrong spot

**Review by vmx — APPROVED** [review]

**Comment by cryptonemo** [comment]
Closing in favor of #196

**Pull request closed without merging**